client/my-sites/site-settings/disconnect-site/index.jsx

```diff
@@ -0,0 +1,64 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import DocumentHead from 'components/data/document-head';
+import FormattedHeader from 'components/formatted-header';
+import Main from 'components/main';
+import PaginationFlow from './pagination-flow';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+
+class DisconnectSite extends Component {
+	componentDidMount() {
+		this.verifySiteIsJetpack();
+	}
+
+	componentDidUpdate() {
+		this.verifySiteIsJetpack();
+	}
+
+	verifySiteIsJetpack() {
+		if ( this.props.siteIsJetpack === false ) {
```

Comment:
We're going to need to [check for AT sites](https://github.com/Automattic/wp-calypso/blob/master/client/state/selectors/is-site-automated-transfer.js) as well, they shouldn't be allowed to disconnect.

Reply:
moved to a HoC: #17555